File: search_pm.js
const fs = require('fs');
const path = require('path');

const keywords = ['purchaseGridColumns', 'selectedCell', 'editingCell', 'productName', '購買日期', '訂購人', '單價', '備註', 'productUrl', '🔗'];

function searchDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git' && file !== 'dist' && file !== 'build') {
        searchDir(fullPath);
      }
    } else {
      if (file.endsWith('.tsx') || file.endsWith('.ts') || file.endsWith('.js') || file.endsWith('.jsx') || file.endsWith('.json') || file.endsWith('.css')) {
        try {
          const content = fs.readFileSync(fullPath, 'utf8');
          for (const kw of keywords) {
            if (content.includes(kw)) {
              console.log(`Found '${kw}' in ${fullPath}`);
            }
          }
        } catch (e) {
          // ignore
        }
      }
    }
  }
}

searchDir('.');

File: search_pm.py
import os

keywords = ['purchaseGridColumns', 'selectedCell', 'editingCell', 'productName', '購買日期', '訂購人', '單價', '備註']

for root, dirs, files in os.walk('.'):
    # prune directories
    dirs[:] = [d for d in dirs if d not in ('node_modules', '.git', 'dist', 'build')]
    for file in files:
        if file.endswith(('.tsx', '.ts', '.js', '.jsx', '.html', '.json', '.css')):
            path = os.path.join(root, file)
            try:
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                    for kw in keywords:
                        if kw in content:
                            print(f"Found '{kw}' in {path}")
            except Exception as e:
                pass
